src/main/java/net/minecraftforge/client/ClientCommandHandler.java

```diff
@@ -0,0 +1,312 @@
+     * treated as client command's that do nothing.
+     * 
+     * {@link net.minecraft.commands.Commands#performCommand(CommandSourceStack, String)} for reference
+     * 
+     * @param sendMessage
+     *            the chat message
+     * @return false leaves the message to be sent to the server, true means it should be caught before {@link LocalPlayer#chat(String)}
+     */
+    public static boolean sendMessage(String sendMessage)
+    {
+        ClientCommandSourceStack source = getSource();
+
+        StringReader reader = new StringReader(sendMessage);
+
+        if (!reader.canRead() || reader.read() != '/')
+        {
+            return false;
+        }
+
+        try
+        {
+            commands.execute(reader, source);
+        }
+        catch (CommandRuntimeException execution)// Probably thrown by the command
+        {
+            Minecraft.getInstance().player.sendMessage(new TextComponent("").append(execution.getComponent()).withStyle(ChatFormatting.RED), Util.NIL_UUID);
+        }
+        catch (CommandSyntaxException syntax)// Usually thrown by the CommandDispatcher
+        {
+            Minecraft.getInstance().player.sendMessage(
+                    new TextComponent("").append(ComponentUtils.fromMessage(syntax.getRawMessage())).withStyle(ChatFormatting.RED), Util.NIL_UUID);
+            if (syntax.getInput() != null && syntax.getCursor() >= 0)
+            {
+                int position = Math.min(syntax.getInput().length(), syntax.getCursor());
+                MutableComponent details = new TextComponent("").withStyle(ChatFormatting.GRAY);
+                details.getStyle().withClickEvent(new ClickEvent(ClickEvent.Action.SUGGEST_COMMAND, reader.getString()));
+                if (position > 10)
+                {
+                    details.append("...");
+                }
+                details.append(syntax.getInput().substring(Math.max(0, position - 10), position));
+                if (position < syntax.getInput().length())
+                {
+                    details.append(new TextComponent(syntax.getInput().substring(position)).withStyle(ChatFormatting.RED, ChatFormatting.UNDERLINE));
+                }
+                details.append(new TranslatableComponent("command.context.here").withStyle(ChatFormatting.RED, ChatFormatting.ITALIC));
+                Minecraft.getInstance().player.sendMessage(new TextComponent("").append(details).withStyle(ChatFormatting.RED), Util.NIL_UUID);
+            }
+        }
+        catch (Exception generic)// Probably thrown by the command
+        {
+            TextComponent message = new TextComponent(generic.getMessage() == null ? generic.getClass().getName() : generic.getMessage());
+            Component component = new TranslatableComponent("command.failed");
+            component.getStyle().withHoverEvent(new HoverEvent(HoverEvent.Action.SHOW_TEXT, message));
```

Comment:
This won't work, as `Style#withHoverEvent` returns a new `Style` (as they are immutable objects).